playwright/integration/connect_wallet.test.ts

```diff
@@ -0,0 +1,26 @@
+import { page, baseURL } from '../utils/test_setup'
+import { account } from '../utils/provider_setup'
+
+describe('Connect Wallet', () => {
+  it('Can connect a Wallet', async () => {
+    // GIVEN: At home page
+    await page.goto(baseURL)
+
+    // WHEN: connected a Wallet
+    // Click text=/.*Connect Wallet.*/
+    await page.click('text=/.*Connect Wallet.*/')
+
+    await page.click('text=Web3')
+
+    // WHEN: Wallet popup is open
+    await page.click('header >> text=Web3')
+
+    // THEN: wallet connected & address available
+    // XPATH: closest ancestor with 0x* text of element with text "Copy address to clipboard"
+    const textWithAddress = await page.textContent(
+      '//*[text()="Copy address to clipboard"]//ancestor::*[starts-with(., "0x")][1]',
+    )
```

Comment:
This is a good example of playwright approach.
As there aren't better ways to distinguish where the address is, instead of `<wallet popup class which is pseudo-random with hash 'cause generated by styled-comps> > text that looks like address`
I opted for `Copy address to clipboard button > closest parent with address`
![dex dev gnosisdev com_](https://user-images.githubusercontent.com/5121491/99369476-846bdc00-28cd-11eb-8c00-f3d56f8866b3.png)

Reply:
I think this is quite thorough, which is good. As opposed to testing for selectors.

What if there a multiple of these components doing the same thing for some reason? page.Textcontent() only returns the first found item? E.g. would be good to test the validity of any other similar/identical item on page.

Reply:
Very well may be
Like [here](https://github.com/gnosis/dex-react/pull/1625#discussion_r524995607)